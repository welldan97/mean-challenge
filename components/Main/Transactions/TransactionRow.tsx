import classNames from 'classnames';
import { memo } from 'react';
import config from '../../../config';
import { Transaction } from '../../../lib/Transaction';

interface Props {
  transaction: Transaction;
}

const TransactionRow = memo(({ transaction }: Props) => {
  if (transaction.type === 'other' && !config.showExtraData) return null;

  return (
    <tr>
      <td
        className={classNames(
          'font-bold',
          transaction.type === 'send' ? 'text-orange-400' : 'text-blue-200',
        )}
      >
        {transaction.type === 'send' && '⇧'}
        {transaction.type === 'receive' && '⇩'}
        {transaction.type === 'other' && '?'}
      </td>
      <td>
        {new Intl.DateTimeFormat('pt-PT', {
          dateStyle: 'short',
          timeStyle: 'short',
        }).format(new Date(transaction.timestamp * 1000))}
      </td>
      <td>{transaction.timestamp}</td>
      <td className="truncate max-w-[100px]">
        {transaction.type === 'send' && transaction.walletAddress}
        {transaction.type === 'receive' && transaction.address}
        {transaction.type === 'other' && transaction.walletAddress}
      </td>
      <td className="truncate max-w-[100px]">
        {transaction.type === 'send' && transaction.address}
        {transaction.type === 'receive' && transaction.walletAddress}
        {transaction.type === 'other' && '-'}
      </td>
      <td>{transaction.type !== 'other' && transaction.amount / 1000000000}</td>
      {config.showExtraData && (
        <>
          <td>{transaction.balanceBefore / 1000000000}</td>
          <td>{transaction.balanceAfter / 1000000000}</td>
        </>
      )}
    </tr>
  );
});

export default TransactionRow;
