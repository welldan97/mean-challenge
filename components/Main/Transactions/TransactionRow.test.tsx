import { render } from '@testing-library/react';

import { transaction } from '../../../lib/test/fixtures/transactions';
import TransactionRow from './TransactionRow';

describe('TransactionRow', () => {
  test('render', () => {
    const { container } = render(
      <table>
        <tbody>
          <TransactionRow transaction={transaction} />
        </tbody>
      </table>,
    );

    expect(container).toMatchSnapshot();
  });
});
