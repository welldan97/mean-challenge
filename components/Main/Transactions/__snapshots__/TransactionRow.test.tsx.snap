// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionRow render 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class="font-bold text-orange-400"
        >
          â‡§
        </td>
        <td>
          05/10/22, 21:34
        </td>
        <td>
          1665005686
        </td>
        <td
          class="truncate max-w-[100px]"
        >
          F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23
        </td>
        <td
          class="truncate max-w-[100px]"
        >
          B1HWEeZCC6Hy3PF24TUmnW8x23oxQirkKwootk27kQTb
        </td>
        <td>
          1
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
